test/e2e/upgrade_management_cluster_test.go

```diff
@@ -2,340 +2,137 @@ package e2e
-			machineList := &clusterv1alpha4.MachineList{}
-			opts := []client.ListOption{
-				client.InNamespace(namespace),
-				client.MatchingLabels{clusterv1.ClusterLabelName: clusterName},
+		isBootstrapProxy := !strings.HasPrefix(clusterProxy.GetName(), "clusterctl-upgrade")
+		if isBootstrapProxy {
+			// remove existing bmh from source and apply first 2 in target
+			Logf("remove existing bmh from source")
+			cmd := exec.Command("bash", "-c", "kubectl delete -f bmhosts_crs.yaml  -n metal3")
+			cmd.Dir = workDir
+			output, err := cmd.CombinedOutput()
+			Logf("Remove existing bmhs:\n %v", string(output))
+			Expect(err).To(BeNil())
+
+			// Apply bmhs [node_0 and node_1] in the management cluster to host the target management cluster
+			for i := 0; i < 2; i++ {
+				applyBMH(i)
 			}
-			g.Expect(bootstrapClusterProxy.GetClient().List(ctx, machineList, opts...)).To(Succeed())
-			for _, machine := range machineList.Items {
-				if machine.Status.GetTypedPhase() == clusterv1alpha4.MachinePhaseRunning {
-					n++
-				}
+		} else {
+			// Apply bmhs [node_2, node_3 and node_4] in the management cluster to host workload cluster
+			for i := 2; i < 5; i++ {
+				applyBMH(i)
 			}
-			g.Expect(n).To(Equal(2))
-		}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Succeed())
-
-		upgradeClusterProxy := bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
-		upgradeClusterClient := upgradeClusterProxy.GetClient()
-
-		By("Initializing the workload cluster with older versions of providers")
-
-		contract := e2eConfig.GetVariable("CAPI_VERSION")
-
-		clusterctl.InitManagementClusterAndWatchControllerLogs(ctx, clusterctl.InitManagementClusterAndWatchControllerLogsInput{
-			ClusterctlBinaryPath:    clusterctlBinaryPath, // use older version of clusterctl to init the management cluster
-			ClusterProxy:            upgradeClusterProxy,
-			ClusterctlConfigPath:    clusterctlConfigPath,
-			CoreProvider:            e2eConfig.GetProviderLatestVersionsByContract(contract, config.ClusterAPIProviderName)[0],
-			BootstrapProviders:      e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmBootstrapProviderName),
-			ControlPlaneProviders:   e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmControlPlaneProviderName),
-			InfrastructureProviders: e2eConfig.GetProviderLatestVersionsByContract(contract, e2eConfig.InfrastructureProviders()...),
-			LogFolder:               filepath.Join(artifactFolder, "clusters", clusterName),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		/*-------------------------------------------------*
-		| Pivot to run ironic/BMO and resources on target |
-		*--------------------------------------------------*/
-		By("Start pivoting")
-		listBareMetalHosts(ctx, bootstrapClusterProxy.GetClient(), client.InNamespace(namespace))
-		listNodes(ctx, upgradeClusterClient)
+		}
+		os.Setenv("PROVIDER_ID_FORMAT", "metal3://{{ ds.meta_data.uuid }}")
```

Comment:
Why is it set here instead of before we start the test? Is it because of the older version? In any case it needs an explanation.

Reply:
if exported before it breaks creating the management because it uses v1beta1 templates

Reply:
WIP comming soon :relaxed:

Reply:
done